wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java

```diff
@@ -1696,4 +1702,41 @@ private String cleanRemoteName(String branchName) throws GitException {
         }
         return returnName;
     }
+
+    /**
+     * Method for generate exception message. The default logic return message from the error.
+     * It also check if the type of the message is for SSL or in case that the error
+     * start with "file name to long" then it raise the relevant message
+     *
+     * @param error
+     *        throwable error
+     * @return exception message
+     */
+    private String generateExceptionMessage(Throwable error) {
+        String message = error.getMessage();
+        //if e caused by an SSLHandshakeException - replace thrown message with a hardcoded message
+        while (error.getCause() != null) {
+            if (error.getCause() instanceof SSLHandshakeException) {
+                message = "The system is not configured to trust the security certificate provided by the Git server";
+                break;
+            } else if (error.getCause() instanceof  IOException) {
+                // Security fix - error message should not include complete local file path on the target system
+                if (message != null && message.startsWith(FILE_NAME_TOO_LONG)) {
+                    try {
+                        String repoPath = repository.getWorkTree().getCanonicalPath();
+                        int startIndex = message.indexOf(repoPath);
+                        int endIndex = message.indexOf(",");
+                        if (startIndex > -1 && endIndex > -1) {
+                            message = FILE_NAME_TOO_LONG + " " + message.substring(startIndex + repoPath.length(), endIndex);
+                        }
```

Comment:
It removed the full path of the file system of the target machine. it is security violation